### PR TITLE
JDK-8312525: New test runtime/os/TestTrimNative.java#trimNative is failing: did not see the expected RSS reduction

### PR DESCRIPTION
**Pull request by tstuefe**
The test causes intermittent rare errors, so far only seen on Oracle 7.9 with glibc 2.17.  

This PR disables, for automated testing, the RSS evaluation part of the test. It retains the mechanical part (testing whether trims happen at all, in the expected number). The stricter variant of the test is kept as manual tests (usable with jtreg without -a).

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8312525](https://bugs.openjdk.org/browse/JDK-8312525): New test runtime/os/TestTrimNative.java#trimNative is failing: did not see the expected RSS reduction (**Bug** - P3)

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Aleksey Shipilev](https://openjdk.org/census#shade) (@shipilev - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14984/head:pull/14984` \
`$ git checkout pull/14984`

Update a local copy of the PR: \
`$ git checkout pull/14984` \
`$ git pull https://git.openjdk.org/jdk.git pull/14984/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14984`

View PR using the GUI difftool: \
`$ git pr show -t 14984`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14984.diff">https://git.openjdk.org/jdk/pull/14984.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14984#issuecomment-1651017464)

### CODE REVIEW

**Review by dholmes-ora — APPROVED** [review]
Seems reasonable. Not 100% sure about the extra logging ... one query in relation to that.

Thanks.

File: src/hotspot/os/linux/os_linux.cpp [hunk 1]
```diff
@@ -4645,6 +4645,19 @@ jint os::init_2(void) {
   log_info(os)("HotSpot is running with %s, %s",
                Linux::libc_version(), Linux::libpthread_version());
 
+  // Log also:
+  // - LD_PRELOAD, since this is the most popular mechanism to divert libc functions
```
[the remark this answers is not in the document]
Comment (tstuefe) [reply src/hotspot/os/linux/os_linux.cpp:4649]: Oh, you are right. I'll check how much more lines that would add; I don't want to bloat the complexity too much.

**Comment by tstuefe** [comment]
@dholmes-ora : I removed parts of the logging. I went down a small rabbit hole about extending to /etc/ld.preload, about unifying that code with existing ld_preload printing, about security concerns (maybe we should obfuscate the values when logging), then I gave up and just removed it, since it does not directly relate to the patch at hand.

**Review by dholmes-ora — APPROVED** [review]
Ok

**Review by shipilev — APPROVED** [review]
Looks fine.

**Comment by tstuefe** [comment]
Thanks @dholmes-ora and @shipilev, and thanks for running this through Oracles CI!

/integrate

**Pull request closed without merging**